astropy/tests/tests/test_imports.py

```diff
@@ -12,25 +12,16 @@
 import astropy
 
 
-def test_imports():
+@pytest.mark.parametrize(
+    "subpkg",
+    [subpkg.name for subpkg in pkgutil.walk_packages(["astropy"]) if subpkg.ispkg],
+)
+def test_imports(subpkg):
     """
-    This just imports all modules in astropy, making sure they don't have any
+    This just imports all top-level sub-packages in astropy, making sure they don't have any
```

Comment:
I don't see a good reason why every single module should be imported individually, especially given that a large fraction of modules are tests which are not supposed to be imported explicitly at all.

Reply:
Users are not supposed to import modules deep within `astropy` internals, so why should there be an explicit test for importing them? Surely if there is a problem in an individual module of some sub-package then trying to import that sub-package will reveal that.